Data-Structures/src/main/java/utilities/AnimalShelter.java

```diff
@@ -0,0 +1,75 @@
+package utilities;
+
+import stacksandqueues.Queue;
+import stacksandqueues.Stack;
+import java.util.Date;
+
+// Implement Animal Shelter using two queues. Each object inserted will contain the actual animal as well as a Date object
+// that keep track of what time the animal got into the Animal Shelter.
+public class AnimalShelter {
+    Queue<Object[]> catQueue;
+    Queue<Object[]> dogQueue;
+
+    public AnimalShelter() {
+        this.catQueue = new Queue<>();
+        this.dogQueue = new Queue<>();
+    }
+
+    // adds animal of either Cat or Dog to Animal Shelter
+    // along with every animal is Date object that time stamp the time the animal was enqueued
+    public void enqueue(Object animal) {

```

Comment:
Using "Object" works since all Java objects are children of the Object class. Alternatively, you could also make a class "Animal" and have Dog and Cat extend Animal. The positive side of doing this is that your method will only accepts Animal type objects.